code/game/objects/items/stacks/stack.dm

```diff
@@ -249,46 +258,71 @@
 	return F
 
 /obj/item/stack/attack_hand(mob/user)
-	if(user.is_in_inactive_hand(src))
-		var/obj/item/stack/F = split(user, 1)
-		user.put_in_hands(F)
+	if(user.is_in_inactive_hand(src) && amount > 1)
+		change_stack(user, 1)
 		if(src && usr.machine == src)
 			spawn(0)
 				interact(usr)
 	else
 		..()
 
+/obj/item/stack/AltClick(mob/living/user)
+	if(!istype(user) || user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+		return
+	if(!in_range(src, user) || issilicon(usr))
+		return
+	if(amount < 1)
+		return
+	//get amount from user
+	var/min = 0
+	var/max = src.get_amount()
+	var/stackmaterial = round(input(user,"How many sheets do you wish to take out of this stack? (Maximum  [max])") as num)
+	if(stackmaterial == null || stackmaterial <= min || stackmaterial >= src.get_amount())
+		return
+	change_stack(user,stackmaterial)
+	to_chat(user, "<span class='notice'>You take [stackmaterial] sheets out of the stack</span>")
+
+/obj/item/stack/proc/change_stack(mob/user,amount)
+	var/obj/item/stack/F = new type(user, amount, FALSE)
+	. = F
+	F.copy_evidences(src)
+	user.put_in_hands(F)
+	add_fingerprint(user)
+	F.add_fingerprint(user)
+	use(amount)
+	zero_amount()
```

Comment:
Won't this be calling `zero_amount()` needlessly as it's already called twice `use()` including after the amount is modified.

Reply:
^ This is pointless, `use` will autodelete the stack if it detects zero_amount triggered.